state/workingset.go

```diff
@@ -178,10 +175,11 @@ func (ws *workingSet) RunActions(
 			return hash.ZeroHash32B, errors.Wrap(err, "failed to convert candidate list to map of cached Candidates")
 		}
 	}
-	if err := ws.handleTsf(tsf); err != nil {
+	tsfs, votes, executions := action.ClassifyActions(actions)
```

Comment:
Need to separate and moving the handling logic into action/<protocol_name>/protocol.go